lib/matplotlib/lines.py

```diff
@@ -1648,7 +1649,7 @@ def __init__(self, line):
             'pick_event', self.onpick)
         self.ind = set()
 
-    canvas = property(lambda self: self.axes.figure.canvas)
+    canvas = property(lambda self: self.axes.get_figure(root=False).canvas)
```

Comment:
```suggestion
    canvas = property(lambda self: self.axes.get_figure(root=True).canvas)
```

Similarly to dpi, canvas always falls back to the root.